phoenix-core-client/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1366,6 +1375,7 @@ private void sendMutations(Iterator<Entry<TableInfo, List<Mutation>>> mutationsI
                     while (itrListMutation.hasNext()) {
                         final List<Mutation> mutationBatch = itrListMutation.next();
                         currentMutationBatch = mutationBatch;
+                        Object[] resultObjects = new Object[mutationBatch.size()];
```

Comment:
I see that we are only reading resultObjects if autocommit is set to true. So we should only allocate this object when autocommit is set. If the autocommit is not set the batch size can be 100s to 1000s and we are just wasting memory by allocating.

Reply:
@jinggou We should only pass the results array when auto-commit is set to true and batch size is 1 since this is the only case we are handling currently. Let's make that explicit in the code.

Reply:
done